src/app/auto-update-handler.ts

```diff
@@ -23,122 +25,136 @@ export class AutoUpdate {
   private autoUpdateTrigger: AutoUpdateTrigger | undefined = undefined;
 
   constructor() {
-    const opts = this.getGenericServerOptions();
-    if (isMac) {
-      this.autoUpdater = new MacUpdater(opts);
-    } else if (isWindowsOS) {
-      this.autoUpdater = new NsisUpdater(opts);
-    }
+    this.getGenericServerOptions().then((opts) => {
```

Comment:
This is only for `Windows` right?

Reply:
yes windows-only

Reply:
Then I think you need to wrap `RegistryStore.getRegistry` and `retrieveWindowsRegistry` inside `isWindowsOS`